codeface/R/cluster/graph_comparison.r

```diff
@@ -91,8 +91,35 @@ vertex.neighborhood.difference <- function(g1, v1, g2, v2) {
   return(difference)
 }
 
+vertex.neighborhood.difference.sym <- function(g1, v1, g2, v2) {
+  ## Calculates the percent similarity using the Jaccard index concept from
+  ## set theory. This considers the neighbour hoods of matching
+  ## verteces from two different graphs and does not consider the edge weights
+  ## rather only the existence of edges.
+  ## -- Output --
+  ## similarity: a percentage of how SIMILAR the neightboods are
+
+  in.v1  <- neighbors(g1, v1, mode="in")
+  out.v1 <- neighbors(g1, v1, mode="out")
+  in.v2  <- neighbors(g2, v2, mode="in")
+  out.v2 <- neighbors(g2, v2, mode="out")
+
+  inout.v1 = union(in.v1,out.v1)
+  inout.v2 = union(in.v2,out.v2)
+
+  totalEdges = length(union(inout.v1,inout.v2))
+  matchEdges = length(intersect(inout.v1,inout.v2))
+
+  if (totalEdges != 0) {
+    difference = 1 - (matchEdges / totalEdges)
+  } else {
+    difference = 0
+  }
+
+  return(difference)
+}
```

Comment:
Because feature graph edges are always both directional I think vertices should be considered equal, when there is an corresponding incomming or outgoing edge.